app/src/main/kotlin/com/wire/android/util/ExifHandler.kt

```diff
@@ -0,0 +1,215 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.android.util
+
+import android.graphics.Bitmap
+import android.graphics.Matrix
+import androidx.exifinterface.media.ExifInterface
+
+/**
+ * Used to remove unnecessary metadata from the image.
+ */
+fun Bitmap.removeExifMetadata(exif: ExifInterface): Bitmap {
+    removableExifAttributes.forEach {
+        exif.setAttribute(it, null)
+    }
+    return this
+}
+
+/**
+ * Rotates the image to its [ExifInterface.ORIENTATION_NORMAL] in case it's rotated with a different orientation than
+ * landscape or portrait See more about exif interface at:
+ * https://developer.android.com/reference/androidx/exifinterface/media/ExifInterface
+ *
+ * @param exif Exif interface for of the image to rotate
+ * @return Bitmap the rotated bitmap or the same in case there is no rotation performed
+ */
+@Suppress("MagicNumber", "TooGenericExceptionCaught")
+fun Bitmap.rotateImageToNormalOrientation(exif: ExifInterface?): Bitmap {
+    val orientation = exif?.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL)
+    val matrix = Matrix()
+    when (orientation) {
+        ExifInterface.ORIENTATION_ROTATE_180 -> matrix.postRotate(180f)
+        ExifInterface.ORIENTATION_ROTATE_90 -> matrix.postRotate(90f)
+        ExifInterface.ORIENTATION_ROTATE_270 -> matrix.postRotate(270f)
+        else -> return this
+    }
+
+    return try {
+        val rotated = Bitmap.createBitmap(this, 0, 0, this.width, this.height, matrix, true)
+        this.recycle()
+        rotated
+    } catch (exception: Exception) {
+        this
+    }
+}
+
+private val removableExifAttributes = arrayOf(
+    // ExifInterface.TAG_ORIENTATION // This is not removed as it's used to rotate the image
+    ExifInterface.TAG_IMAGE_WIDTH,
+    ExifInterface.TAG_IMAGE_LENGTH,
```

Comment:
do need to remove TAG_IMAGE_WIDTH and TAG_IMAGE_LENGTH if it can lead to any rendering issues?

Reply:
Done ! That's a good point, those are rewritten anyway as the image is scaled down for profile pictures. But it is better to not remove in case we use this function later for other purposes.